### PR TITLE
Feature/website tutorial

### PR DESCRIPTION
**Pull request by gary-archer**
Suggested updates to reduce code level plumbing and make the example feel like a business API.

The focus of the code example is productive integration tests, rather than the API code itself:
- Using a JWT library to create mock access tokens
- Using Wiremock as a tool to host the JSON Web KeySet

### CODE REVIEW

File: src/test/java/se/curity/examples/spark/integration/AbstractApiAuthorizationTest.java [hunk 1]
```diff
@@ -23,67 +32,52 @@
...
+        if (started) {
+            return;
         }
-    }
 
-    @BeforeEach
-    void publishJwks() {
-        wm1.stubFor(get("/jwks")
+        started = true;
+        var options = new WireMockConfiguration().port(8443).httpDisabled(false);
```
Comment (ju-cu) [inline src/test/java/se/curity/examples/spark/integration/AbstractApiAuthorizationTest.java:68]: no need to specify `httpDisabled(false)`?

**Review by ju-cu — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,90 +1,78 @@
...
-In addition, the example includes tests to demonstrate how to effectively work
-with authorization and JWTs during development.
+A simple Java REST API implemented using [Spark Java](http://sparkjava.com).\
+The API uses an OAuth filter to implement its JWT validation on every request.\
+The API then uses claims to implement its business  authorization.
 
-The application is a simple API created with [Spark](http://sparkjava.com),
-but it could be built using any framework which supports Java Servlets. 
-It includes hardcoded data to visualize the results of the authorization rules.
+The API's integration tests use a JWT library to create a keypair.\
```
[the remark this answers is not in the document]
Comment (gary-archer) [reply README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -1,90 +1,78 @@
...
-The endpoint `/api/products` returns a list of products depending on the `country` claim in the access token. The list may be empty. 
-The endpoint `/api/products/<1-5>` returns the product details if the user has a valid (non-empty) `subscription_level` claim. Exclusive products (`2`,`5`) require a `premium` subscription. 
-Checkout the [Working With Claims Tutorial](https://curity.io/resources/learn/working-with-claims/) for how to configure claims for access tokens in the Curity Identity Server.
 
-If the user is not authorized to access the resource, i.e. the JWT is missing a valid subscription level or the user tries to access a product from a different country, then the server will return `403`. 
-If the resource cannot be found, e.g. the product with the given ID does not exist, then the server will return `404`. 
-Have a look at `se/curity/examples/products/ProductServiceMapImpl.java` for the details of the provided example data.
+Tests spin up a keypair and use the private key to issue JWT access tokens for testing.\
+Wiremock is used to expose the JSON Web KeySet at http://localhost:8443/oauth/v2/oauth-anonymous/jwks. \
```
[the remark this answers is not in the document]
Comment (gary-archer) [reply README.md]: fixed

File: src/test/java/se/curity/examples/spark/integration/AbstractApiAuthorizationTest.java (hunk 1 above)
Comment (gary-archer) [reply src/test/java/se/curity/examples/spark/integration/AbstractApiAuthorizationTest.java]: removed - better

File: src/main/java/se/curity/examples/spark/ServerOptions.java [hunk 4]
```diff
@@ -161,11 +123,7 @@ public ServerOptions(@Nullable String[] args) {
 
                 switch (argumentName) {
                     case "--jwksurl" -> {
-                        try {
-                            this.jwksUrl = new URL(argumentValue);
-                        } catch (MalformedURLException exception) {
-                            throw new IllegalArgumentException(String.format("Invalid value for JWKS URL: %s", exception.getMessage()));
-                        }
+                        this.jwksUrl = argumentValue;
```
[the remark this answers is not in the document]
Comment (gary-archer) [reply src/main/java/se/curity/examples/spark/ServerOptions.java]: replaced it

**Pull request merged**